h2o-core/src/main/java/water/udf/CDistributionFunc.java

```diff
@@ -0,0 +1,86 @@
+package water.udf;
+
+import java.io.Serializable;
+
+/**
+ * Custom Distribution Function
+ */
+public interface CDistributionFunc extends CFunc, Serializable {
+
+    /**
+     * Canonical link
+     * @param f value in original space, to be transformed to link space
+     * @return link(f)
+     */
+    public double link(double f);
+
+    /**
+     * Canonical link inverse
+     * @param f value in link space, to be transformed back to original space
+     * @return linkInv(f)
+     */
+    public double linkInv(double f);
+
+    /**
+     * String version of link inverse (for POJO scoring code generation)
+     * @param f value to be transformed by link inverse
+     * @return String that turns into compilable expression of linkInv(f)
+     */
+    public String linkInvString(String f);
+
+    /**
+     * Deviance of given distribution function at predicted value f
+     * @param w observation weight
+     * @param y (actual) response
+     * @param f (predicted) response in original response space (including offset)
+     * @return deviance
+     */
+    public double deviance(double w, double y, double f);
```

Comment:
only needed for metrics, imho

Reply:
Yes, it is true, however, how do you want to calculate deviance if a user defined a custom distribution only for the gradient calculation?

Reply:
That is a good point. My thinking was that it is only used for metrics for regression models, so maybe it is one thing we could make optional/move to another class. But I think that is too early to decide. Please continue with your POC and don't worry - I am just thinking out loud.